File: ClassLibraryCisepro/CONTABILIDAD/BANCOS/ClassBancos.vb
Imports System.Data
Imports System.Data.SqlClient
Imports ClassLibraryCisepro.ENUMS
Imports ClassLibraryCisepro.ProcesosSql

Namespace CONTABILIDAD.BANCOS
    Public Class ClassBancos
        ''' <summary>
        ''' ATRIBUTOS QUE ESTAN LIGADOS A LA(S) TABLA(S) CORRESPONDIENTE(S)
        ''' SI AGREGO UNA COLUMNA EN LA TABLA (BD), ES RECOMENDABLE CREAR SU ATRIBUTO CORRESPONDIENTE
        ''' CON EL MISMO TIPO DE DATO
        ''' TODOS LOS METODOS DE GUARDAR, SELECCIONAR, BUSCAR, ESTAN LIGADOS A UN PROCEDIMIENTO EN BD Y SU TIPO DE RETRNO GENERAL ES DATATABLE</summary>
        Public IdBancos As Integer
        Public Codigo As String
        Public Nombre As String
        Public Direccion As String
        Public OficialCredito As String
        Public Telefono As String
        Public Fax As String
        Public Email As String
        Public Estado As Integer


        Public Function SeleccionarRegistrosBancos(ByVal tipoCon As TipoConexion) As DataTable
            Return ComandosSql.SeleccionarQueryToDataTable(tipoCon, "SeleccionarRegistrosBancos", True)
        End Function

        Public Function SeleccionarOrdenadosRegistrosBancos(ByVal tipoCon As TipoConexion) As DataTable
            Return ComandosSql.SeleccionarQueryToDataTable(tipoCon, "SeleccionarOrdenadosRegistrosBancos", True)
        End Function

        Public Function BuscarMayorIdBancos(ByVal tipoCon As TipoConexion) As Integer
            Dim data = ComandosSql.SeleccionarQueryToDataTable(tipoCon, "BuscarMayorIdBancos", True)
            Return If(data.Rows.Count = 0, 0, If(IsDBNull(data.Rows(0)(0)), String.Empty, CStr(data.Rows(0)(0))))
        End Function

        Public Function BuscarIdBancoXNombreBanco(ByVal tipoCon As TipoConexion, ByVal nombreBancoBuscado As String) As Integer
            Dim pars = New List(Of Object())
            pars.Add(New Object() {"NOMBRE_BANCO", SqlDbType.NVarChar, nombreBancoBuscado})
            Dim data = ComandosSql.SeleccionarQueryWithParamsToDataTable(tipoCon, "buscarIdBancoXNombreBanco", True, pars)
            Return If(data.Rows.Count = 0, 0, If(IsDBNull(data.Rows(0)(0)), String.Empty, CStr(data.Rows(0)(0))))
        End Function

        Public Function BuscarCodigoBancoXNombreBanco(ByVal tipoCon As TipoConexion, ByVal nombreBancoBuscado As String) As String
            Dim pars = New List(Of Object())
            pars.Add(New Object() {"NOMBRE_BANCO", SqlDbType.NVarChar, nombreBancoBuscado})
            Dim data = ComandosSql.SeleccionarQueryWithParamsToDataTable(tipoCon, "buscarCodigoBancoXNombreBanco", True, pars)
            Return If(data.Rows.Count = 0, String.Empty, If(IsDBNull(data.Rows(0)(0)), String.Empty, CStr(data.Rows(0)(0))))
        End Function

        Public Function NuevoRegistroBancos(ByVal tipoCon As TipoConexion) As Boolean
            Dim comando = New SqlCommand
            With comando
                .CommandType = CommandType.StoredProcedure
                .CommandText = "NuevoRegistroBancos"
                .Parameters.AddWithValue("@ID_BANCO", SqlDbType.Int).Value = IdBancos
                .Parameters.AddWithValue("@CODIGO_BANCO", SqlDbType.NVarChar).Value = Codigo
                .Parameters.AddWithValue("@NOMBRE_BANCO", SqlDbType.NVarChar).Value = Nombre
                .Parameters.AddWithValue("@DIRECCION_BANCO", SqlDbType.NVarChar).Value = Direccion
                .Parameters.AddWithValue("@OFICIAL_CREDITO_BANCO", SqlDbType.NVarChar).Value = OficialCredito
                .Parameters.AddWithValue("@TELEFONO_BANCO", SqlDbType.NVarChar).Value = Telefono
                .Parameters.AddWithValue("@FAX_BANCO", SqlDbType.NVarChar).Value = Fax
                .Parameters.AddWithValue("@EMAIL_BANCO", SqlDbType.NVarChar).Value = Email
                .Parameters.AddWithValue("@ESTADO_BANCO", SqlDbType.Int).Value = Estado
            End With
            Return ComandosSql.ProcesarComandoCisepro(tipoCon, comando)
        End Function

        Public Function ModificarRegistroBancos(ByVal tipoCon As TipoConexion) As Boolean
            Dim comando = New SqlCommand
            With comando
                .CommandType = CommandType.StoredProcedure
                .CommandText = "modificarRegistroBancos"
                .Parameters.AddWithValue("@ID_BANCO", SqlDbType.Int).Value = IdBancos
                .Parameters.AddWithValue("@NOMBRE_BANCO", SqlDbType.NVarChar).Value = Nombre
                .Parameters.AddWithValue("@DIRECCION_BANCO", SqlDbType.NVarChar).Value = Direccion
                .Parameters.AddWithValue("@OFICIAL_CREDITO_BANCO", SqlDbType.NVarChar).Value = OficialCredito
                .Parameters.AddWithValue("@TELEFONO_BANCO", SqlDbType.NVarChar).Value = Telefono
                .Parameters.AddWithValue("@FAX_BANCO", SqlDbType.NVarChar).Value = Fax
                .Parameters.AddWithValue("@EMAIL_BANCO", SqlDbType.NVarChar).Value = Email
                .Parameters.AddWithValue("@ESTADO_BANCO", SqlDbType.Int).Value = Estado
            End With
            Return ComandosSql.ProcesarComandoCisepro(tipoCon, comando)
        End Function

    End Class
End Namespace

File: ClassLibraryCisepro/CONTABILIDAD/BANCOS/ClassBancos.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using Microsoft.VisualBasic.CompilerServices;

namespace ClassLibraryCisepro.CONTABILIDAD.BANCOS
{
    public class ClassBancos
    {
        /// <summary>
        /// ATRIBUTOS QUE ESTAN LIGADOS A LA(S) TABLA(S) CORRESPONDIENTE(S)
        /// SI AGREGO UNA COLUMNA EN LA TABLA (BD), ES RECOMENDABLE CREAR SU ATRIBUTO CORRESPONDIENTE
        /// CON EL MISMO TIPO DE DATO
        /// TODOS LOS METODOS DE GUARDAR, SELECCIONAR, BUSCAR, ESTAN LIGADOS A UN PROCEDIMIENTO EN BD Y SU TIPO DE RETRNO GENERAL ES DATATABLE</summary>
        public int IdBancos;
        public string Codigo;
        public string Nombre;
        public string Direccion;
        public string OficialCredito;
        public string Telefono;
        public string Fax;
        public string Email;
        public int Estado;


        public DataTable SeleccionarRegistrosBancos(ClassLibraryCisepro.ENUMS.TipoConexion tipoCon)
        {
            return ClassLibraryCisepro.ProcesosSql.ComandosSql.SeleccionarQueryToDataTable(tipoCon, "SeleccionarRegistrosBancos", true);
        }

        public DataTable SeleccionarOrdenadosRegistrosBancos(ClassLibraryCisepro.ENUMS.TipoConexion tipoCon)
        {
            return ClassLibraryCisepro.ProcesosSql.ComandosSql.SeleccionarQueryToDataTable(tipoCon, "SeleccionarOrdenadosRegistrosBancos", true);
        }

        public int BuscarMayorIdBancos(ClassLibraryCisepro.ENUMS.TipoConexion tipoCon)
        {
            var data = ClassLibraryCisepro.ProcesosSql.ComandosSql.SeleccionarQueryToDataTable(tipoCon, "BuscarMayorIdBancos", true);
            return Conversions.ToInteger(data.Rows.Count == 0 ? 0 : data.Rows[0][0] is DBNull ? string.Empty : Conversions.ToString(data.Rows[0][0]));
        }

        public int BuscarIdBancoXNombreBanco(ClassLibraryCisepro.ENUMS.TipoConexion tipoCon, string nombreBancoBuscado)
        {
            var pars = new List<object[]>();
            pars.Add(new object[] { "NOMBRE_BANCO", SqlDbType.NVarChar, nombreBancoBuscado });
            var data = ClassLibraryCisepro.ProcesosSql.ComandosSql.SeleccionarQueryWithParamsToDataTable(tipoCon, "buscarIdBancoXNombreBanco", true, pars);
            return Conversions.ToInteger(data.Rows.Count == 0 ? 0 : data.Rows[0][0] is DBNull ? string.Empty : Conversions.ToString(data.Rows[0][0]));
        }

        public string BuscarCodigoBancoXNombreBanco(ClassLibraryCisepro.ENUMS.TipoConexion tipoCon, string nombreBancoBuscado)
        {
            var pars = new List<object[]>();
            pars.Add(new object[] { "NOMBRE_BANCO", SqlDbType.NVarChar, nombreBancoBuscado });
            var data = ClassLibraryCisepro.ProcesosSql.ComandosSql.SeleccionarQueryWithParamsToDataTable(tipoCon, "buscarCodigoBancoXNombreBanco", true, pars);
            return data.Rows.Count == 0 ? string.Empty : data.Rows[0][0] is DBNull ? string.Empty : Conversions.ToString(data.Rows[0][0]);
        }

        public bool NuevoRegistroBancos(ClassLibraryCisepro.ENUMS.TipoConexion tipoCon)
        {
            var comando = new SqlCommand();
            comando.CommandType = CommandType.StoredProcedure;
            comando.CommandText = "NuevoRegistroBancos";
            comando.Parameters.AddWithValue("@ID_BANCO", SqlDbType.Int).Value = IdBancos;
            comando.Parameters.AddWithValue("@CODIGO_BANCO", SqlDbType.NVarChar).Value = Codigo;
            comando.Parameters.AddWithValue("@NOMBRE_BANCO", SqlDbType.NVarChar).Value = Nombre;
            comando.Parameters.AddWithValue("@DIRECCION_BANCO", SqlDbType.NVarChar).Value = Direccion;
            comando.Parameters.AddWithValue("@OFICIAL_CREDITO_BANCO", SqlDbType.NVarChar).Value = OficialCredito;
            comando.Parameters.AddWithValue("@TELEFONO_BANCO", SqlDbType.NVarChar).Value = Telefono;
            comando.Parameters.AddWithValue("@FAX_BANCO", SqlDbType.NVarChar).Value = Fax;
            comando.Parameters.AddWithValue("@EMAIL_BANCO", SqlDbType.NVarChar).Value = Email;
            comando.Parameters.AddWithValue("@ESTADO_BANCO", SqlDbType.Int).Value = Estado;
            return ClassLibraryCisepro.ProcesosSql.ComandosSql.ProcesarComandoCisepro(tipoCon, comando);
        }

        public bool ModificarRegistroBancos(ClassLibraryCisepro.ENUMS.TipoConexion tipoCon)
        {
            var comando = new SqlCommand();
            comando.CommandType = CommandType.StoredProcedure;
            comando.CommandText = "modificarRegistroBancos";
            comando.Parameters.AddWithValue("@ID_BANCO", SqlDbType.Int).Value = IdBancos;
            comando.Parameters.AddWithValue("@NOMBRE_BANCO", SqlDbType.NVarChar).Value = Nombre;
            comando.Parameters.AddWithValue("@DIRECCION_BANCO", SqlDbType.NVarChar).Value = Direccion;
            comando.Parameters.AddWithValue("@OFICIAL_CREDITO_BANCO", SqlDbType.NVarChar).Value = OficialCredito;
            comando.Parameters.AddWithValue("@TELEFONO_BANCO", SqlDbType.NVarChar).Value = Telefono;
            comando.Parameters.AddWithValue("@FAX_BANCO", SqlDbType.NVarChar).Value = Fax;
            comando.Parameters.AddWithValue("@EMAIL_BANCO", SqlDbType.NVarChar).Value = Email;
            comando.Parameters.AddWithValue("@ESTADO_BANCO", SqlDbType.Int).Value = Estado;
            return ClassLibraryCisepro.ProcesosSql.ComandosSql.ProcesarComandoCisepro(tipoCon, comando);
        }

    }
}